File: week-01/vacuum-cleaner-route/vacuum_cleaner_route_test.go
package vacuumcleanerroute

import "testing"

func TestIsCompleteRoute(t *testing.T) {
	tests := []struct {
		name string
		in   string
		want bool
	}{
		{
			name: "horizontal",
			in:   "LR",
			want: true,
		},
		{
			name: "non-complete route",
			in:   "URURD",
			want: false,
		},
		{
			name: "all direction",
			in:   "RUULLDRD", want: true,
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := IsCompleteRoute(tt.in); got != tt.want {
				t.Errorf("IsCompleteRoute(%s) = %v, want %v", tt.in, got, tt.want)
			}
		})
	}
}

File: week-01/vacuum-cleaner-route/vacuum_cleaner_route_test.py
from vacuum_cleaner_route import is_complete_route


def test_is_complete_route():
    tests = [
        {
            "moves": "",
            "want": True,
        },
        {
            "moves": "LR",
            "want": True,
        },
        {
            "moves": "RUULLDRD",
            "want": True,
        },
        {
            "moves": "URURD",
            "want": False,
        },
    ]

    for test in tests:
        assert is_complete_route(test["moves"]) == test["want"]
